app/src/ui/changes/changes-list.tsx

```diff
@@ -193,27 +193,29 @@ export class ChangesList extends React.Component<
     this.props.onDiscardAllChanges(this.props.workingDirectory.files)
   }
 
-  private onDiscardChanges = (paths: string | string[]) => {
+  private onDiscardChanges = (files: ReadonlyArray<string>) => {
     const workingDirectory = this.props.workingDirectory
 
-    if (paths instanceof Array) {
-      const files: WorkingDirectoryFileChange[] = []
-      paths.forEach(path => {
-        const file = workingDirectory.files.find(f => f.path === path)
-        if (file) {
-          files.push(file)
+    if (files.length === 1) {
+      const modifiedFile = workingDirectory.files.find(f => f.path === files[0])
+
+      return modifiedFile && this.props.onDiscardChanges(modifiedFile)
```

Comment:
I'm not the greatest fan of the implicit `null` comparison here as well as the `return void`. Could we make this a bit easier to read?

```ts
if (modifiedFile != null) {
  this.props.onDiscardChanges(modifiedFile)
}
```